test/evergreen.yml

```diff
@@ -136,6 +136,33 @@ functions:
           DEFINED_EVERGREEN_CONFIG_FLAGS=${DEFINED_EVERGREEN_CONFIG_FLAGS/\-DHAVE_BUILTIN_EXTENSION_ZSTD=1/}
         fi
 
+        if [ "${build_variant|}" = "macos-1100" ]; then
+          # For mac builds, we want explicitly tell cmake which python to use, as
+          # well as the matching library directory and header files. The find_libpython
+          # module gives us the library.
+          SYSPY=${python_binary}
+          $SYSPY -mvenv venv
+          source venv/bin/activate
+          pip install find_libpython
+          SYSPYLIB=`find_libpython`
+          SYSPYINCDEF=
+
+          # We have the shared library to link to, it may be named simply 'Python3'.
+          # If that's the case, use the associated dylib symlink found in an expected relative
+          # location. Also get the location of the header files. We'll give this all to cmake.
+          if [ $(basename $SYSPYLIB) = 'Python3' ]; then
```

Comment:
Another future-proofing comment: Is there a possibility that the library would be just called 'Python' in another variant? It's like that on my computer, but then I have Python installed from a different source. Again, no need to change this if this is what we have on that OS image.

Reply:
Fixed.  I liked your future proofing!